tests/utils/test_enumerablemap.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starkware_utils.error_handling import StarkException
+from starkware.starknet.definitions.error_codes import StarknetErrorCode
+from utils import assert_revert, contract_path, TRUE, FALSE
+
+@pytest.fixture
+async def enumerablemap_mock():
+    starknet = await Starknet.empty()
+
+    enumerablemap = await starknet.deploy(
+        contract_path("tests/mocks/enumerablemap_mock.cairo")
+    )
+
+    return enumerablemap
+
+
+@pytest.mark.asyncio
+async def test_set(enumerablemap_mock):
+    enumerablemap = enumerablemap_mock
+
+    success = await enumerablemap.set(6, 8).invoke()
+    assert success.result.success == TRUE
+
+    executed_info = await enumerablemap.contains(6).call()
+    assert executed_info.result == (TRUE,)
```

Comment:
shouldn't we also test the value and length are correct?

Reply:
i figured that was tested in the values test but i can add it

Reply:
which one? `test_update`? to me this is the one testing `set`, including its behavior. `update` has another purpose.